Add pr template

neil-hickey opened the pull request:
Adds a pull request template to the repo

Fixes #388

neil-hickey committed:
Add PR template

- This will help streamline the release notes - the PR now requests a
  release note be added
- PR authors are now guided to our contribution / dev guides

Fixes: https://github.com/vmware-tanzu/carvel-kapp-controller/issues/388

Signed-off-by: Neil Hickey <nhickey@vmware.com>

aaronshurley commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,44 @@
+#### What type of PR is this?
+
+<!--
+Please add a label to your pull request that matches the type of change you are making.
```
I think we'll want to change this wording since non-maintainers can't add labels ([GH docs](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#applying-a-label)).

danielhelfand replied:
I think removing this section for now makes sense. It would work better if we had an approach like Kubernetes does with prow and the `/` actions in comments. Something to consider for future.

neil-hickey replied:
agreed. Removed for now, it can always be added later as the project matures.

aaronshurley reviewed (changes requested):
Overall I think this is a great addition. I just had one nit.

joe-kimmel-vmw commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,44 @@
+<!--  Thanks for sending a pull request!  Here are some tips for you:
+
+1. If this is your first time, please read our contributor guidelines: https://github.com/vmware-tanzu/carvel-kapp-controller/blob/develop/CONTRIBUTING.md and developer guide https://github.com/vmware-tanzu/carvel-kapp-controller/blob/develop/docs/dev.md
```
nit but if there isn't a `2.` do we need the `1.` ?

neil-hickey replied:
the purpose of the 1. was so if you thought of another, it could be added. If the review finds nothing to add, I shall remove it! :) 👍

neil-hickey replied:
sounds like we don't have any others... so I will remove the `1.`. :)

joe-kimmel-vmw reviewed (commented):
is the vision that we'll all do this every time?  How many of these sections are optional?

neil-hickey commented:
> is the vision that we'll all do this every time? How many of these sections are optional?

Do you feel like any of them should be optional? I think we will find over time if its useful or 90% of the time everyone ignores a certain section. Github doesn't give you a "required" to be filled in section in PRs that I am aware of. 

What was the concern around asking if we would have to do this every time?

danielhelfand commented on .github/PULL_REQUEST_TEMPLATE.md:
````diff
@@ -0,0 +1,44 @@
+-->
+Fixes #
+
+#### Does this PR introduce a user-facing change?
+<!--
+If no, just write "NONE" in the release-note block below.
+If yes, a release note is required:
+Enter your extended release note in the block below. 
+
+-->
+```release-note
+
+```
````
Can we make use of this? I guess I worry about having this now as there is no guarantee we will go through each pull request to get this release note. If there is some goal of including automation, I think that makes sense.

neil-hickey replied:
Yeh, I wonder if / how it can be part of githubs new automatically generated release notes work.

The reason I put this here was that it gives us the option of saying, hey this thing breaks alot / adds a major thing and we want to call it out. As opposed to every small change needs a release note.

Yes its a higher burden to releasing, but if we release every 3 months with 20+PRs, and we have to trawl through all these PRs I think we have a different issue.

neil-hickey committed:
Remove 'what type of pr' section

- This may be more valuable in the future if we have automation around
  the tags / labels

Signed-off-by: Neil Hickey <nhickey@vmware.com>

neil-hickey committed:
Remove `1.` from tips section

- We settled on just one tip, so no need for a numbered list!

Signed-off-by: Neil Hickey <nhickey@vmware.com>

aaronshurley reviewed (approved):
LGTM

danielhelfand reviewed (approved):
lgtm
